### PR TITLE
Fixing yolov8-seg validation

### PR DESCRIPTION
**Pull request by abhinavnmagic**
The PR enables segmentation evaluation for yolov8 models, which currently fails. It implements a function that extends `detach` to PyTorch expected types ie. list or tuples.

Testing:
- Evaluation works for both detection and segmentation models obtained form either SparseML or ultraltyics

### CODE REVIEW

File: src/sparseml/yolov8/utils/helpers.py [hunk 1]
```diff
@@ -69,3 +71,14 @@ def create_grad_sampler(
         / train_loader.batch_size,
     )
     return grad_sampler
+
+
+def detach(x: Union[torch.Tensor, List, Tuple]):
```
Comment (natuan) [inline src/sparseml/yolov8/utils/helpers.py:76]: Would it be better to move this func into sparseml/pytorch/utils/helpers.py so it could be reused more?

**Review by natuan — CHANGES_REQUESTED** [review]

File: src/sparseml/yolov8/utils/helpers.py (hunk 1 above)
Comment (abhinavnmagic) [reply src/sparseml/yolov8/utils/helpers.py:76]: Yes, definitely. If we believe it would reused, that's a better place.

**Review by natuan — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]
